### ISSUE TITLE
Erroneous arithmetic operation with HOEXTRAP BCs

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/MAESTROeX#92]
Compile `TEST_PROBLEMS/test_stability` with 
```
make USE_OMP=FALSE DEBUG=TRUE -j 20
```
Then run as:
```
./Maestro2d.gnu.DEBUG.ex inputs_2d_C_regression  amrex.fpe_trap_invalid=1
```
It aborts during the initial projections with:
```
Writing plotfile pltInitData after InitData
inner sponge: r_sp      , r_tp      :   0.960937500000E+10    0.960937500000E+10
 
Time to write plotfile: 0.847286215
inner sponge: r_sp      , r_tp      :   0.960937500000E+10    0.960937500000E+10
 
Doing initial projection
Erroneous arithmetic operation
See Backtrace.0 file for details
```

The backtrace is attached.
[Backtrace.0.txt](https://github.com/AMReX-Astro/MAESTROeX/files/3513240/Backtrace.0.txt)

**Comment by ajnonaka** [comment]
This is a corner ghost cell issue when you have 2 physical boundaries.  If you assume no ghost cells are initialized but the valid domain is filled, the loop that fills ghost cells tries to fill the corner ghost cell by extrpolating the interior values... but for that corner ghost cell those "interior" values are ghost cells in another direction.  One way to alleviate the problem is to initialize all ghost cells to some value; but there are several places this needs to be done.

**Comment by ajnonaka** [comment]
Fixed with 8910306341c
This may not fix every MAESTRO issue with physical boundaries, but this example runs now.
